src/osd/OSD.cc

```diff
@@ -5619,8 +5619,17 @@ bool OSD::_is_healthy()
   }
 
   if (is_waiting_for_healthy()) {
+     utime_t now = ceph_clock_now();
+     utime_t grace = utime_t(cct->_conf->osd_max_markdown_period, 0);
+     while (!osd_markdown_log.empty() &&
+             osd_markdown_log.front() + grace < now)
+       osd_markdown_log.pop_front();
+     if (osd_markdown_log.size() <= 1) {
+       dout(5) << __func__ << " first time marked as down,"
+               << " try reboot unconditionally" << dendl;
```

Comment:
I don't understand this comment. What's going to do a reboot? I guess you're referring to the caller in ```OSD::start_boot()```, but that's not necessarily a reboot and it's not fair to assume all callers will have that same behavior.

Reply:
The only caller is ```OSD::start_boot()```.

Reply:
Yeah, but it's generally a bad idea to insert dependencies on the parent call chain into utility functions like this.